lib/internal/modules/esm/loader.js

```diff
@@ -374,14 +377,18 @@ class ESMLoader {
       return module;
     };
     const ModuleJob = require('internal/modules/esm/module_job');
-    const job = new ModuleJob(
-      this, url, undefined, evalInstance, false, false);
-    this.moduleMap.set(url, undefined, job);
-    const { module } = await job.run();
+    const namespace = await onLoad.tracePromise(async () => {
+      const job = new ModuleJob(
+        this, url, undefined, evalInstance, false, false);
+      this.moduleMap.set(url, undefined, job);
+
+      const { module } = await job.run();
+      return module.getNamespace();
+    }, { parentURL: '<eval>', url });
```

Comment:
This blocks allocates quite a few new promises for something that's very hot.
I think we can avoid some of it by doing `return job.run()`.

Reply:
It needs to return module.getNamespace() or the event.result value received by diagnostics_channel will be the module container rather than the module itself, which is not intended to public API surface.